### ISSUE TITLE
[Feature] Add test case for checking consistency of policy spec between CUE and Golang

### ISSUE THREAD

**Issue by chivalryq** [kubevela/kubevela#5507]
For now we have some policy spec in golang stored in `apis/core.oam.dev/v1alpha1`. The corresponding CUE spec are stored in `vela-templates/definitions/internal/policy`. 

Although these files are trying to describe the same specifications like `apis/core.oam.dev/v1alpha1/readonly_policy_types.go` and `vela-templates/definitions/internal/policy/read-only.cue`, there is no source of truth. None of them is generated from each other. So we need some test cases to check if they are consistent.

------
we need some test case to cover

_Originally posted by @wonderflow in https://github.com/kubevela/kubevela/pull/5498#pullrequestreview-1296976654_

**Comment by swastik959** [comment]
@chivalryq can you please tell me the location to where the test case is to be made

**Comment by rhzx3519** [comment]
Please provide some clues about how to verify the consistency between cue and go. Thank you.

**Comment by chivalryq** [comment]
Hi @rhzx3519 @swastik959 This is a kinda tricky problem. There are two ways to do that.
1. Make one is generated from another.
2. Use some mock data to test it fits both the Go struct and CUE schema.

Considering the generating problems I think it's easier take the second method to implement this. You can add test in `apis/core.oam.dev/v1alpha1`

**Comment by rhzx3519** [comment]
> Hi @rhzx3519 @swastik959 This is a kinda tricky problem. There are two ways to do that.
> 
> 1. Make one is generated from another.
> 2. Use some mock data to test it fits both the Go struct and CUE schema.
> 
> Considering the generating problems I think it's easier take the second method to implement this. You can add test in `apis/core.oam.dev/v1alpha1`

Yep, I agree the second method would be an easy way to implement. I'll take this issue.

**Comment by swastik959** [comment]
@rhzx3519 sorry but I have already started working on this issue

**Comment by swastik959** [comment]
@chivalryq I have tried to make a example test case please check if this is what you want

**Comment by briankane** [comment]
This issue has been inactive for over 2 years and is scheduled for closure as part of our backlog maintenance.

  **If this issue is still relevant:**
  - Please comment below to keep it open
  - Provide updated context if the original description is outdated

  **Timeline:** If there's no activity by **Friday, October 24th**, this issue will be automatically closed.

  **Note:** Issues can always be reopened if they become relevant again. We encourage the community to reopen and reprioritize any incorrectly 
  closed issues.

**Comment by briankane** [comment]
This issue has been closed following the stale issue notice. 

If this issue is still relevant and should remain open, please:
1. Reopen the issue
2. Remove the 'stale' label
3. Provide an update on the current status or importance

Thank you for your understanding and contribution to KubeVela!
